api/admin/cli/main.go

```diff
@@ -0,0 +1,311 @@
+package main
+
+import (
+	"crypto/rand"
+	"encoding/base32"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/wissance/Ferrum/managers/redis"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/wissance/Ferrum/api/admin/cli/operations"
+	"github.com/wissance/Ferrum/data"
+	"github.com/wissance/Ferrum/logging"
+	sf "github.com/wissance/stringFormatter"
+)
+
+type Manager interface {
+	GetRealm(realmName string) (*data.Realm, error)
+	GetClient(realmName string, clientName string) (*data.Client, error)
+	GetUser(realmName string, userName string) (data.User, error)
+
+	CreateRealm(newRealm data.Realm) error
+	CreateClient(realmName string, clientNew data.Client) error
+	CreateUser(realmName string, userNew data.User) error
+
+	DeleteRealm(realmName string) error
+	DeleteClient(realmName string, clientName string) error
+	DeleteUser(realmName string, userName string) error
+
+	UpdateRealm(realmName string, realmNew data.Realm) error
+	UpdateClient(realmName string, clientName string, clientNew data.Client) error
+	UpdateUser(realmName string, userName string, userNew data.User) error
+
+	SetPassword(realmName string, userName string, password string) error
+}
+
+const defaultConfig = "./config.json"
+
+var (
+	argConfigFile  = flag.String("config", "", "Path relative to the application binary file")
+	argOperation   = flag.String("operation", "", "")
+	argResource    = flag.String("resource", "", "")
+	argResource_id = flag.String("resource_id", "", "")
+	argParams      = flag.String("params", "", "This is the name of the realm for operations on client or user resources")
+	argValue       = flag.String("value", "", "Json object")
+)
+
+func main() {
+	flag.Parse()
+	var manager Manager
+	{
+		pathToConfigFile, err := createPathToConfig(*argConfigFile, defaultConfig)
+		if err != nil {
+			log.Fatalf("createPathToConfig failed: %s", err)
+		}
+		cfg, err := readConfig(pathToConfigFile)
+		if err != nil {
+			log.Fatalf("readConfig failed: %s", err)
+		}
+
+		logger := logging.CreateLogger(&cfg.Logging)
+		redisManager, err := redis.CreateRedisDataManager(&cfg.DataSource, logger)
```

Comment:
Here we should create Common `DataContext` because it will be easier to extend functionality in future when new Data storage type will be added